src/runner/mod.rs

```diff
@@ -594,4 +594,21 @@ mod tests {
         }
     }
 
+    mod impl_visitor_block_for_runner {
+        use super::*;
+
+        use block::*;
+        use header::*;
+
+        #[test]
+        fn it_can_be_called() {
+            // arrange
+            let runner = Runner::default();
+            let block = Block::Example(Example::new(ExampleHeader::default(), |_| ExampleReport::Success));
```

Comment:
seems like a bug to me (or at least a regression).
This wont compile:
```rust
let block = Block::Example(Example::new(ExampleHeader::default(), |_| ()));
```
It should !

Reply:
Same for `false`:
```
/Users/thomas/.cargo/bin/cargo test --color=always --package rspec --lib runner::tests::impl_visitor_block_for_runner -- --nocapture
   Compiling rspec v1.0.0-beta.4 (file:///Users/thomas/projects/rust/rspec)
error[E0308]: mismatched types
   --> src/runner/mod.rs:607:83
    |
607 |             let block = Block::Example(Example::new(ExampleHeader::default(), |_| false));
    |                                                                                   ^^^^^ expected enum `report::example::ExampleReport`, found bool
    |
    = note: expected type `report::example::ExampleReport`
               found type `bool`
```

Reply:
No, this is correct. `Example::new(…)` takes a closure which handles the conversion from `()`/`bool`/… to `ExampleReport`. One could use `F: 'static + Fn(&T) -> impl Into<ExampleReport>` instead of `F: 'static + Fn(&T) -> ExampleReport` in future, but that's not available yet. Also `Example::new(…)` is not intended for public use anyway. That's what `ctx.then(…)` is for.

Reply:
okay I see

Reply:
However if you see a way to make this cleaner I'm all for it. :)